File: src/de/hsh/alexander/src/actor/level_elements/Fan.java
package de.hsh.alexander.src.actor.level_elements;

import common.actor.LevelElement;

import static de.hsh.alexander.src.actor.ResourcePaths.Actor.LevelElements.Fan.pictures;

public class Fan extends LevelElement {

    private static final double default_delay = 3;
    private static final double scale         = 3.4;

    public Fan( double x, double y ) {
        this( x, y, scale );
    }

    public Fan( double x, double y, double scale ) {
        super( x, y, pictures );
        this.setSwitchingDelay( default_delay );
        this.scaleImage( scale );
    }
}

File: Games/Alex/src/de/hsh/alexander/actor/level_elements/Fan.kt
package de.hsh.alexander.actor.level_elements

import common.actor.LevelElement
import de.hsh.alexander.actor.ResourcePaths

class Fan @JvmOverloads constructor(x: Double, y: Double, scale: Double = this.scale) :
        LevelElement(x, y, *ResourcePaths.Actor.LevelElements.Fan.pictures) {
    companion object {
        private const val default_delay = 3.0
        private const val scale = 3.4
    }

    init {
        this.switchingDelay = (default_delay)
        scaleImage(scale)
    }
}
